### PR TITLE
2.x.x lambda todos

### PR DESCRIPTION
**Pull request by adam-fowler**
- Update TODOs lambda sample for v2
- Update scripts to use archive build plugin for packaging and AWS SAM for deployment.

### CODE REVIEW

File: todos-lambda/scripts/build-and-package.sh [hunk 1]
```diff
@@ -13,20 +13,7 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-set -eu
+here=$(dirname "$0")
 
-base=$(pwd)
```
Comment (Joannis) [inline todos-lambda/scripts/build-and-package.sh:18]: One thing we add to our build scripts is a check that the user is on ARM64

**Review by Joannis — APPROVED** [review]
LGTM, two small comments

File: todos-lambda/scripts/build-and-package.sh (hunk 1 above)
Comment (adam-fowler) [reply todos-lambda/scripts/build-and-package.sh:18]: What's your actual check?

Comment (Joannis) [reply todos-lambda/scripts/build-and-package.sh:18]: In our case we enforce AMD64, due to us having had some issues specific to Lambda runtime on ARM64:

```sh
if [ "$(uname -m)" = "arm64" ]; then
  echo "Error: Unsupported architecture. Please build from an AMD64 CPU or use the GitHub CI"
  exit 1
fi
```

File: todos-lambda/Package.swift [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
-    products: [
-        .executable(name: "HummingbirdTodosLambda", targets: ["App"]),
+        .macOS(.v14),
     ],
     dependencies: [
-        .package(url: "https://github.com/hummingbird-project/hummingbird.git", from: "1.0.0"),
-        .package(url: "https://github.com/hummingbird-project/hummingbird-lambda.git", from: "1.0.0-rc.3"),
-        .package(url: "https://github.com/soto-project/soto.git", from: "6.0.0"),
+        .package(url: "https://github.com/hummingbird-project/hummingbird.git", from: "2.0.0-alpha.1"),
+        .package(url: "https://github.com/hummingbird-project/hummingbird-lambda.git", branch: "2.x.x-request-headers"),
```
[the remark this answers is not in the document]
Comment (adam-fowler) [reply todos-lambda/Package.swift]: done

File: todos-lambda/scripts/build-and-package.sh (hunk 1 above)
Comment (adam-fowler) [reply todos-lambda/scripts/build-and-package.sh:18]: Added architecture check

**Pull request merged**